index.js

```diff
@@ -5,7 +5,16 @@ var isglob = require('is-glob');
 var pathDirname = require('path-dirname');
 
 module.exports = function globParent(str) {
-	str += 'a'; // preserves full path in case of trailing path separator
+	// replace braces/brackets sections with *
+	str = str.replace(/(^|[^\\])(\{([^{}]*?)}|\[([^\[\]]*?)\])/g, '$1*');
+	if (/[\}\}]$/.test(str)) str += '/';
```

Comment:
Yeah, this won't work, none of the implementations require both braces to be escaped for the brace pattern to be escaped. Also, this fails on nested braces when only one of them is escaped.